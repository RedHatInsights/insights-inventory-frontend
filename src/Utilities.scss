// Sass Imports

@import 'Utilities/_variables.scss';
@import 'Utilities/_colors.scss';
@import 'Utilities/_helpers.scss';
@import 'Utilities/_mixins.scss';

@import '~@redhat-cloud-services/frontend-components/index.css';
