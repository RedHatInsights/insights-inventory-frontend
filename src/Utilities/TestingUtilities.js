import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

export const mountWithRouter = (Component, initialEntries) => {
  const wrapper = mount(
    <MemoryRouter initialEntriest={initialEntries}>{Component}</MemoryRouter>
  );

  return wrapper;
};

export const TestWrapper = ({
  children,
  routerProps = { initialEntries: ['/'] },
  path,
  store = {},
}) => {
  return (
    <MemoryRouter {...routerProps}>
      <Provider store={store}>
        {path ? (
          <Routes>
            <Route path={path} element={children} />
          </Routes>
        ) : (
          children
        )}
      </Provider>
    </MemoryRouter>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.any.isRequired,
  routerProps: PropTypes.shape({
    initialEntries: PropTypes.arrayOf(PropTypes.string),
  }),
  path: PropTypes.string,
  store: PropTypes.any,
};
