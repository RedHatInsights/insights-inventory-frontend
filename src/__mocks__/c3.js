// eslint-disable-next-line no-undef
module.exports = () => 'c3';

