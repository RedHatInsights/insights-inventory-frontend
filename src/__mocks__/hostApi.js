import { hosts, tags } from '../api/api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios, { onNoMatch: 'throwException' });
export const mockTags = new MockAdapter(tags.axios, { onNoMatch: 'throwException' });
