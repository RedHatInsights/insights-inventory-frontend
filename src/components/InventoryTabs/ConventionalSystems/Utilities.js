import flatMap from 'lodash/flatMap';
import {
  HOST_GROUP_CHIP,
  RHCD_FILTER_KEY,
  UPDATE_METHOD_KEY,
} from '../../../Utilities/constants';

const mapTags = ({ category, values }) =>
  values.map(
    ({ tagKey, value }) =>
      `${category ? `${category}/` : ''}${tagKey}${value ? `=${value}` : ''}`
  );

const filterMapper = {
  staleFilter: ({ staleFilter }, searchParams) =>
    staleFilter.forEach((item) => searchParams.append('status', item)),
  osFilter: ({ osFilter }, searchParams) => {
    // TODO This is very hackish. There should be a schema/feature on how the values translate into params
    const osParams = Object.entries(
      Object.values(osFilter).reduce((acc, item) => ({ ...acc, ...item }), {})
    )
      .filter(([, value]) => value === true)
      .map(([key]) => key);

    osParams.forEach((item) => {
      const keyParts = item.split('-');
      const [osName, value] = [
        keyParts.slice(0, keyParts.length - 2).join(' '),
        keyParts[keyParts.length - 1],
      ];

      if (value && osName) {
        searchParams.append('operating_system', `${osName}${value}`);
      }
    });
  },
  registeredWithFilter: ({ registeredWithFilter }, searchParams) =>
    registeredWithFilter?.forEach((item) =>
      searchParams.append('source', item)
    ),
  value: ({ value, filter }, searchParams) =>
    value === 'hostname_or_id' &&
    Boolean(filter) &&
    searchParams.append('hostname_or_id', filter),
  tagFilters: ({ tagFilters }, searchParams) =>
    tagFilters?.length > 0 &&
    searchParams.append('tags', flatMap(tagFilters, mapTags)),
  rhcdFilter: ({ rhcdFilter }, searchParams) =>
    rhcdFilter?.forEach((item) => searchParams.append(RHCD_FILTER_KEY, item)),
  lastSeenFilter: ({ lastSeenFilter }, searchParams) =>
    Object.keys(lastSeenFilter || {})?.forEach(
      (item) =>
        item === 'mark' &&
        searchParams.append('last_seen', lastSeenFilter[item])
    ),
  updateMethodFilter: ({ updateMethodFilter }, searchParams) =>
    updateMethodFilter?.forEach((item) =>
      searchParams.append(UPDATE_METHOD_KEY, item)
    ),
  hostGroupFilter: ({ hostGroupFilter }, searchParams) =>
    hostGroupFilter?.forEach((item) =>
      searchParams.append(HOST_GROUP_CHIP, item)
    ),
};

// TODO It might be better to not use a SearchParam instance and mutate it
// This should rather just return a plain object that can be
// applied with react routers useSearch params.
export const calculateFilters = (searchParams, filters = []) => {
  filters.forEach((filter) => {
    Object.keys(filter).forEach((key) => {
      filterMapper?.[key]?.(filter, searchParams);
    });
  });
  return searchParams;
};

export const calculatePagination = (searchParams, page, perPage) => {
  const currSearch = new URLSearchParams(location.search);
  const newPage = page !== undefined ? page : currSearch.get('page');
  const newPerPage =
    perPage !== undefined ? perPage : currSearch.get('per_page');
  !isNaN(parseInt(newPage)) && searchParams.append('page', newPage);
  !isNaN(parseInt(newPerPage)) && searchParams.append('per_page', newPerPage);
};
