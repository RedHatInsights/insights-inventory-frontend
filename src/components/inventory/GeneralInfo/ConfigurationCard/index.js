export { default } from './ConfigurationCard';
export * from './ConfigurationCard';
