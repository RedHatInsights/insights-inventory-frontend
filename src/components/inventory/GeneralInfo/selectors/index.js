export { default } from './selectors';
export * from './selectors';
