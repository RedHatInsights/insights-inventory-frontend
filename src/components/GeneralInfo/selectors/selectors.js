/* eslint-disable camelcase */
import { registered } from '../../../Utilities/index';
import { verifyCollectorStaleness } from '../../../Utilities/sharedFunctions';
function safeParser(toParse, key) {
  try {
    return JSON.parse(toParse);
  } catch {
    return { [key]: toParse };
  }
}

export const propertiesSelector = (
  {
    number_of_cpus,
    number_of_sockets,
    cores_per_socket,
    ramSize,
    disk_devices,
    sap_sids,
    system_purpose,
    cpu_flags,
  } = {},
  { facts } = {}
) => ({
  cpuNumber: number_of_cpus || facts?.rhsm?.CPU_CORES,
  sockets: number_of_sockets || facts?.rhsm?.CPU_SOCKETS,
  coresPerSocket:
    cores_per_socket ||
    (facts?.rhsm?.CPU_CORES &&
      facts?.rhsm?.CPU_CORES &&
      Number(facts?.rhsm?.CPU_CORES, 10) /
        Number(facts?.rhsm?.CPU_SOCKETS, 10)),
  ramSize: ramSize || (facts?.rhsm?.MEMORY && `${facts?.rhsm?.MEMORY} GB`),
  storage:
    disk_devices &&
    disk_devices.map(({ device, label, mount_point, options, type }) => ({
      ...(device && safeParser(device, 'device')),
      label,
      ...(mount_point && safeParser(mount_point, 'mountpoint')),
      ...(options && safeParser(options, 'options')),
      ...(type && safeParser(type, 'mounttype')),
    })),
  sapIds: sap_sids,
  systemPurpose: system_purpose?.usage,
  cpuFlags: cpu_flags,
});

export const operatingSystem = (
  {
    arch,
    operating_system,
    os_kernel_version,
    last_boot_time,
    kernel_modules,
    system_update_method,
  } = {},
  { facts } = {}
) => ({
  release: operating_system,
  kernelRelease: os_kernel_version,
  architecture: arch || facts?.rhsm?.ARCHITECTURE,
  bootTime: last_boot_time,
  kernelModules: kernel_modules,
  systemUpdateMethod: system_update_method,
});

export const biosSelector = ({
  bios_vendor,
  bios_version,
  bios_release_date,
} = {}) => ({
  vendor: bios_vendor,
  version: bios_version,
  releaseDate: bios_release_date,
});

export const bootcSelector = ({ bootc_status } = {}) => ({
  bootedImage: bootc_status?.booted?.image,
  bootedImageDigest: bootc_status?.booted?.image_digest,
  stagedImage: bootc_status?.staged?.image,
  stagedImageDigest: bootc_status?.staged?.image_digest,
  rollbackImage: bootc_status?.rollback?.image,
  rollbackImageDigest: bootc_status?.rollback?.image_digest,
});

export const infrastructureSelector = (
  {
    infrastructure_type,
    infrastructure_vendor,
    public_ipv4_addresses,
    public_dns,
    network = {},
  } = {},
  { facts } = {}
) => ({
  type:
    infrastructure_type ||
    (facts?.rhsm?.IS_VIRTUAL !== undefined &&
      (facts?.rhsm?.IS_VIRTUAL ? 'virtual' : 'physical')) ||
    undefined,
  vendor: infrastructure_vendor,
  public_ipv4_addresses: public_ipv4_addresses,
  ipv4: network.ipv4,
  ipv6: network.ipv6,
  nics: network.interfaces,
  fqdn: public_dns,
});

export const configurationSelector = ({
  installed_packages,
  enabled_services,
  running_processes,
  repositories,
} = {}) => ({
  packages: installed_packages,
  services: enabled_services,
  processes: running_processes,
  repositories,
});

export const collectionInformationSelector = ({
  insights_client_version,
  insights_egg_version,
} = {}) => ({
  client: insights_client_version,
  egg: insights_egg_version,
});

export const getCollectorStatus = (collectorStaleness) => {
  return collectorStaleness
    ? verifyCollectorStaleness(collectorStaleness) !== 'Fresh'
      ? 'Stale'
      : 'Active'
    : 'N/A';
};

export const getDefaultCollectors = (entity) =>
  registered
    ?.filter((reporter) => reporter.label !== 'insights-client not connected')
    .map((reporter) => ({
      name: reporter.label,
      status: getCollectorStatus(
        entity?.per_reporter_staleness[reporter.value]
      ),
      updated: entity?.per_reporter_staleness[reporter.value]?.last_check_in,
      details: {
        name: reporter.idName,
        id: entity?.[reporter.idValue],
      },
    }));

export const systemStatus = ({ stale_timestamp } = {}) => ({
  stale: new Date() > new Date(stale_timestamp),
});
