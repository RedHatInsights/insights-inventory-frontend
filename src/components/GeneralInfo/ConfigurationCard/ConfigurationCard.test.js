/* eslint-disable camelcase */
import React from 'react';
import ConfigurationCard from './ConfigurationCard';
import configureStore from 'redux-mock-store';
import { configTest } from '../../../__mocks__/selectors';
import { renderWithRouter } from '../../../Utilities/TestingUtilities';
import { useParams } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { screen, waitFor } from '@testing-library/react';

const location = {};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => location,
  useParams: jest.fn(() => ({
    modalId: 'testModal',
  })),
}));

describe('ConfigurationCard', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          ...configTest,
        },
      },
    };
    location.pathname = 'localhost:3000/example/path';
  });

  it('should render correctly - no data', () => {
    const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
    const view = renderWithRouter(<ConfigurationCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with data', () => {
    const store = mockStore(initialState);
    const view = renderWithRouter(<ConfigurationCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render enabled/disabled', () => {
    const store = mockStore({
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          ...configTest,
          repositories: {
            enabled: [{}],
            disabled: [{}],
          },
        },
      },
    });
    const view = renderWithRouter(<ConfigurationCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  describe('api', () => {
    it('should NOT call handleClick', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      renderWithRouter(<ConfigurationCard store={store} />);

      await userEvent.click(screen.getAllByRole('link')[0]);
      await waitFor(() => {
        expect(onClick).not.toHaveBeenCalled();
      });
    });

    it('should call handleClick on packages', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/installed_packages';
      useParams.mockImplementation(() => ({ modalId: 'installed_packages' }));
      renderWithRouter(
        <ConfigurationCard handleClick={onClick} store={store} />
      );

      await userEvent.click(screen.getAllByRole('link')[0]);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalled();
      });
    });

    it('should call handleClick on services', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/services';
      useParams.mockImplementation(() => ({ modalId: 'services' }));
      renderWithRouter(
        <ConfigurationCard handleClick={onClick} store={store} />
      );

      await userEvent.click(screen.getAllByRole('link')[1]);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalled();
      });
    });

    it('should call handleClick on processes', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/running_processes';
      useParams.mockImplementation(() => ({ modalId: 'running_processes' }));
      renderWithRouter(
        <ConfigurationCard handleClick={onClick} store={store} />
      );

      await userEvent.click(screen.getAllByRole('link')[2]);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalled();
      });
    });

    it('should call handleClick on repositories', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/repositories';
      useParams.mockImplementation(() => ({ modalId: 'repositories' }));
      renderWithRouter(
        <ConfigurationCard handleClick={onClick} store={store} />
      );

      await userEvent.click(screen.getAllByRole('link')[3]);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalled();
      });
    });
  });

  [
    ['hasPackages', 'Installed packages'],
    ['hasServices', 'Services'],
    ['hasProcesses', 'Running processes'],
    ['hasRepositories', 'Repositories'],
  ].map(([flag, title]) =>
    it(`should not render ${title}`, () => {
      const store = mockStore(initialState);
      renderWithRouter(
        <ConfigurationCard store={store} {...{ [flag]: false }} />
      );
      expect(screen.queryByText(title)).toBeNull();
    })
  );

  it('should render extra', () => {
    const store = mockStore(initialState);
    const view = renderWithRouter(
      <ConfigurationCard
        store={store}
        extra={[
          { title: 'something', value: 'test' },
          {
            title: 'with click',
            value: '1 tests',
            onClick: (_e, handleClick) => handleClick('Something', {}, 'small'),
          },
        ]}
      />
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
});
