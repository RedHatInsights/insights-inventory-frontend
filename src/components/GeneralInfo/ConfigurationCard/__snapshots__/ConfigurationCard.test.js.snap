// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationCard api should NOT call handleClick 1`] = `
<Connect(ConfigurationCardCore)
  detailLoaded={false}
  extra={Array []}
  handleClick={[Function]}
  hasPackages={true}
  hasProcesses={true}
  hasRepositories={true}
  hasServices={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConfigurationCardCore
    configuration={
      Object {
        "packages": Array [
          "packages",
        ],
        "processes": Array [
          "processes",
        ],
        "repositories": Object {
          "disabled": Array [],
          "enabled": Array [
            Object {
              "base_url": "test-url",
              "enabled": true,
              "gpgcheck": true,
              "name": "test-name",
            },
          ],
        },
        "services": Array [
          "services",
        ],
      }
    }
    detailLoaded={true}
    dispatch={[Function]}
    extra={Array []}
    handleClick={[Function]}
    hasPackages={true}
    hasProcesses={true}
    hasRepositories={true}
    hasServices={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <LoadingCard
      isLoading={false}
      items={
        Array [
          Object {
            "onClick": [Function],
            "singular": "package",
            "target": "installed_packages",
            "title": "Installed packages",
            "value": 1,
          },
          Object {
            "onClick": [Function],
            "singular": "service",
            "target": "services",
            "title": "Services",
            "value": 1,
          },
          Object {
            "onClick": [Function],
            "plural": "processes",
            "singular": "process",
            "target": "running_processes",
            "title": "Running processes",
            "value": 1,
          },
          Object {
            "onClick": [Function],
            "target": "repositories",
            "title": "Repositories",
            "value": "1 enabled",
          },
        ]
      }
      title="Configuration"
    >
      <Card>
        <article
          className="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-4"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe={true}
          id=""
        >
          <CardBody>
            <div
              className="pf-c-card__body"
            >
              <Stack
                hasGutter={true}
              >
                <div
                  className="pf-l-stack pf-m-gutter"
                >
                  <StackItem>
                    <div
                      className="pf-l-stack__item"
                    >
                      <TextContent>
                        <div
                          className="pf-c-content"
                        >
                          <Text
                            component="h1"
                          >
                            <h1
                              className=""
                              data-ouia-component-id="OUIA-Generated-Text-4"
                              data-ouia-component-type="PF4/Text"
                              data-ouia-safe={true}
                              data-pf-content={true}
                            >
                              Configuration
                            </h1>
                          </Text>
                        </div>
                      </TextContent>
                    </div>
                  </StackItem>
                  <StackItem
                    isFilled={true}
                  >
                    <div
                      className="pf-l-stack__item pf-m-fill"
                    >
                      <TextContent>
                        <div
                          className="pf-c-content"
                        >
                          <TextList
                            component="dl"
                          >
                            <dl
                              className=""
                            >
                              <TextListItem
                                component="dt"
                              >
                                <dt
                                  className=""
                                >
                                  Installed packages
                                </dt>
                              </TextListItem>
                              <TextListItem
                                component="dd"
                              >
                                <dd
                                  className=""
                                >
                                  <Clickable
                                    onClick={[Function]}
                                    singular="package"
                                    target="installed_packages"
                                    value={1}
                                  >
                                    <a
                                      href="http://localhost:5000//installed_packages"
                                      onClick={[Function]}
                                    >
                                      1 package
                                    </a>
                                  </Clickable>
                                </dd>
                              </TextListItem>
                              <TextListItem
                                component="dt"
                              >
                                <dt
                                  className=""
                                >
                                  Services
                                </dt>
                              </TextListItem>
                              <TextListItem
                                component="dd"
                              >
                                <dd
                                  className=""
                                >
                                  <Clickable
                                    onClick={[Function]}
                                    singular="service"
                                    target="services"
                                    value={1}
                                  >
                                    <a
                                      href="http://localhost:5000//services"
                                      onClick={[Function]}
                                    >
                                      1 service
                                    </a>
                                  </Clickable>
                                </dd>
                              </TextListItem>
                              <TextListItem
                                component="dt"
                              >
                                <dt
                                  className=""
                                >
                                  Running processes
                                </dt>
                              </TextListItem>
                              <TextListItem
                                component="dd"
                              >
                                <dd
                                  className=""
                                >
                                  <Clickable
                                    onClick={[Function]}
                                    plural="processes"
                                    singular="process"
                                    target="running_processes"
                                    value={1}
                                  >
                                    <a
                                      href="http://localhost:5000//running_processes"
                                      onClick={[Function]}
                                    >
                                      1 process
                                    </a>
                                  </Clickable>
                                </dd>
                              </TextListItem>
                              <TextListItem
                                component="dt"
                              >
                                <dt
                                  className=""
                                >
                                  Repositories
                                </dt>
                              </TextListItem>
                              <TextListItem
                                component="dd"
                              >
                                <dd
                                  className=""
                                >
                                  <Clickable
                                    onClick={[Function]}
                                    target="repositories"
                                    value="1 enabled"
                                  >
                                    <a
                                      href="http://localhost:5000//repositories"
                                      onClick={[Function]}
                                    >
                                      1 enabled
                                    </a>
                                  </Clickable>
                                </dd>
                              </TextListItem>
                            </dl>
                          </TextList>
                        </div>
                      </TextContent>
                    </div>
                  </StackItem>
                </div>
              </Stack>
            </div>
          </CardBody>
        </article>
      </Card>
    </LoadingCard>
  </ConfigurationCardCore>
</Connect(ConfigurationCardCore)>
`;

exports[`ConfigurationCard should not render hasPackages 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-9"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-9"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should not render hasProcesses 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-11"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-11"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should not render hasRepositories 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-12"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-12"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should not render hasServices 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-10"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-10"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should render correctly - no data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should render correctly with data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-2"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-2"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should render enabled/disabled 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-3"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-3"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled / 1 disabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`ConfigurationCard should render extra 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-13"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-13"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Installed packages
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 package
              </a>
            </dd>
            <dt
              class=""
            >
              Services
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//services"
              >
                1 service
              </a>
            </dd>
            <dt
              class=""
            >
              Running processes
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 process
              </a>
            </dd>
            <dt
              class=""
            >
              Repositories
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
            <dt
              class=""
            >
              something
            </dt>
            <dd
              class=""
            >
              test
            </dd>
            <dt
              class=""
            >
              with click
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//undefined"
              >
                1 tests
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;
